app/controllers/articles_controller.rb

```diff
@@ -46,6 +46,16 @@ def destroy
     redirect_to articles_url, notice: 'Article was successfully destroyed.'
   end
 
+  def like
+    set_article
```

Comment:
don't need to call this. use the before_action callback and ask it to run `set_article` for this action also.